kernel/src/main/scala-2.12/cats/kernel/compat/scalaVersionSpecific.scala

```diff
@@ -50,4 +51,8 @@ private[cats] object scalaVersionSpecific {
       that: T
     )(implicit w1: A => TraversableLike[El1, Repr1], w2: T => IterableLike[El2, Repr2]) = (a, that).zipped
   }
+
+  implicit class setExtension[A](private val s: java.util.Set[A]) extends AnyVal {
+    def asScala: mutable.Set[A] = `set asScala`(s)
+  }
```

Comment:
This extension method is only used for `Gen.oneOf` in tests, and the latter only requires `Iterable`. So I'd suggest to go with the minimally required type here as well:
```suggestion
  implicit class iterableExtension[A](private val s: java.lang.Iterable[A]) extends AnyVal {
    def asScala: Iterable[A] = iterableAsScalaIterable(s) // assuming import scala.collection.JavaConverters._
  }
```

Reply:
Besides, if this extension method is for the sake of testing only, I wonder if it'd make sense to move it from "main" to "test" scope?